### ISSUE TITLE
Determining UTxO set size in the UTxO HD era

### ISSUE THREAD

**Issue by deepfire** [input-output-hk/ouroboros-network#3973]
Since the UTxO is no longer entirely stored in memory, we're losing direct means of estimation of its size.

Yet, we still need this information, in the long run -- for performance estimation/prediciton reasons.

**Comment by nfrisby** [comment]
Yeah, it should be easy enough to add a `Word64` counter that gets included in some log `Tracer` events.

**Comment by jorisdral** [comment]
@dnadales, @Jasagredo and I discussed solutions to providing the size of the full UTxO. We identified two axes along which possible solutions can be placed:

* Should it be a query or not?
* Should this consult the backing store in IO (and use the changelog), or should we keep a counter in memory?  

There are several considerations
* Conceptually, retrieving the full UTxO size would make sense as a query, but that might not mesh well with [how performance metrics are gathered in the node](https://github.com/input-output-hk/cardano-node/blob/master/cardano-node/src/Cardano/Node/Tracing/Tracers/StartLeadershipCheck.hs#L59), where the internals of the ChainDB are accessed to retrieve the metrics.
* Since the UTxO is on-disk, it would make sense to add functionality to the LedgerDB to retrieve this information from disk. This would make retrieving the size an IO action.
* Keeping a counter on the in-memory parts of the ledger state could be achieved if we update the counter each time we apply a block to the ledger state. This might not be a future-proof solution however, since it could convolute the ledger state and block application, specially if we would want to keep track of additional counters in the future (i.e., delegation map size). Furthermore, the ledger does not need this information
* Keeping a counter in the LedgerDB changelog would also be a possiblity, and would remove the need for ledgerstates to keep track of the counter.

Possible solutions (in order of viability):
1. Keep track of a counters on the LedgerDB changelog
2. Retrieve the UTxO size from disk through the backing store interface
3. Keep track of a counter on ledger states that is updated each time a block is applied.
4. Provide full UTxO size as a query
